src/main/scala/org/scalajs/tools/tsimporter/Importer.scala

```diff
@@ -296,13 +296,15 @@ class Importer(val output: java.io.PrintWriter) {
       anyAsDynamic: Boolean = false): TypeRef = {
 
     tpe.name match {
-      case "any"     => if (anyAsDynamic) TypeRef.Dynamic else TypeRef.Any
-      case "dynamic" => TypeRef.Dynamic
-      case "void"    => TypeRef.Unit
-      case "number"  => TypeRef.Double
-      case "bool"    => TypeRef.Boolean
-      case "boolean" => TypeRef.Boolean
-      case "string"  => TypeRef.String
+      case "any"       => if (anyAsDynamic) TypeRef.Dynamic else TypeRef.Any
+      case "dynamic"   => TypeRef.Dynamic
+      case "void"      => TypeRef.Unit
+      case "number"    => TypeRef.Double
+      case "bool"      => TypeRef.Boolean
+      case "boolean"   => TypeRef.Boolean
+      case "string"    => TypeRef.String
+      case "null"      => TypeRef.Null
+      case "undefined" => TypeRef.Null
```

Comment:
The `undefined` case is not correct. It should be `TypeRef.Unit`. In Scala.js, `undefined` and `()` (of type `Unit`) are the same thing. `null` is a different thing, which is `null` of type `Null`.